ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/StateRef.java

```diff
@@ -74,21 +70,17 @@ public int getIndex() {
      */
     public static StateRef parse(@NotNull final String value, DigestService digestService) {
         try {
-            if (!value.contains(DELIMITER)) {
-                throw new IllegalArgumentException(
-                        MessageFormat.format("Failed to parse a StateRef from the specified value. The value: {0} doesn't contain expected delimiter (:)", value)
-                        );
-            }
-            if (value.length() < MINIMUM_HASH_LENGTH) {
+            final int lastIndexOfDelimiter = value.lastIndexOf(DELIMITER);
```

Comment:
Unnecessary variable as it's only used once.

Reply:
Isn't this variable used in three places? 
``` kotlin
final int lastIndexOfDelimiter = value.lastIndexOf(DELIMITER);
            if (lastIndexOfDelimiter == -1) {
                throw new IllegalArgumentException(
                        MessageFormat.format("Failed to parse a StateRef from the specified value. The delimiter (:) is missing: {0}.", value)
                );
            }
            final String subStringBeforeDelimiter = value.substring(0, lastIndexOfDelimiter);
            final String subStringAfterDelimiter = value.substring(lastIndexOfDelimiter + 1);
```